### DISCUSSION TITLE
TreeViewItem Selected when TextBox within has focus

### DISCUSSION THREAD

**Discussion by cristinathoughtpennies** [AvaloniaUI/Avalonia#9698] [General]
Hello! I can't figure out how to get the functionality where a TreeViewItem is selected when a control in the item (specifically the text box is giving me trouble) has focus. Any help would be appreciated!

This is the closest I've gotten, but it will select ALL the parent TreeViewItems, and then stop working when an item itself is selected
```
<TreeView.Styles>
<Style Selector="TreeViewItem[IsKeyboardFocusWithin=True]">
	<Setter Property="IsSelected" Value="True"/>
</Style>
</TreeView.Styles>
```

**Comment by timunie** [comment]
I think a Behavior could be an option which can be attached to TreeView. Check out Avalonia.Behaviors repo for basic samples.

**Reply by cristinathoughtpennies** [reply]
Think I was able to figure it out, thanks!
This worked for me as far as i can tell when placed in the TreeDataTemplate
```
<Interaction.Behaviors>
	<EventTriggerBehavior EventName="GotFocus" SourceObject="{Binding #NameOfTextBox}">
		<ChangePropertyAction TargetObject="{Binding $parent[TreeView]}" PropertyName="SelectedItem" Value="{Binding}"/>
	</EventTriggerBehavior>
	<EventTriggerBehavior EventName="LostFocus" SourceObject="{Binding #NameOfTextBox}">
		<ChangePropertyAction TargetObject="{Binding $parent}" PropertyName="IsSelected" Value="False"/>
	</EventTriggerBehavior>
</Interaction.Behaviors>
```
